src/VisualStudio/Core/Def/Telemetry/FileLogger.cs

```diff
@@ -113,4 +93,24 @@ public void LogBlockEnd(FunctionId functionId, LogMessage logMessage, int unique
 
     private void LogBlockEvent(FunctionId functionId, LogMessage logMessage, int uniquePairId, string blockEvent)
         => Log(functionId, $"[{blockEvent} - {uniquePairId}] {logMessage.GetMessage()}");
+
+    private ValueTask ProcessWorkQueueAsync(
+        ImmutableSegmentedList<(FunctionId functionId, string message)> list, CancellationToken cancellationToken)
+    {
+        foreach (var (functionId, message) in list)
+            _buffer.AppendLine($"{DateTime.Now} ({functionId}) : {message}");
+
+        IOUtilities.PerformIO(() =>
+        {
+            if (!File.Exists(_logFilePath))
+            {
+                Directory.CreateDirectory(PathUtilities.GetDirectoryName(_logFilePath));
+            }
+
+            File.AppendAllText(_logFilePath, _buffer.ToString());
+            _buffer.Clear();
```

Comment:
Looks like it's trying to do a single write, instead of N smaller writes. I have no issue keeping this.

Reply:
I kind of like Jason's suggestion too, it can still be a single call into the File API, something like:

```
            var now = DateTime.Now;
            File.AppendAllLines(_logFilePath, list.Select((functionId, message) => $"{now} ({functionId}) : {message}"));
```

Reply:
Can we separate out core parts of PRs, vs nice to haves?   I'm fine with potentially making this change in a future PR, or having one of you guys do it.  The core goal here is simplifying the threading primitives.  I don't want to be iterating a lot on making extra changes that take away from that work.  

Please separate out nits and suggestions, from requirements for a PR to go in.  Thanks @jasonmalinowski @ToddGrun

Reply:
@CyrusNajmabadi Even if we don't want to optimize, at least move _buffer to a local. Otherwise we're leaving a field around that doesn't need to be a field, and looks like there's _more_ complexity than there is.

Reply:
I can make this change in a followup PR.

Reply:
Done in 052c7eee3a5